core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -72,9 +72,11 @@ private[spark] case class ExecutorDeadException(message: String)
 /**
  * Exception thrown when Spark returns different result after upgrading to a new version.
  */
-private[spark] class SparkUpgradeException(version: String, message: String, cause: Throwable)
-  extends RuntimeException("You may get a different result due to the upgrading of Spark" +
-    s" $version: $message", cause)
+private[spark] class SparkUpgradeException(
```

Comment:
I'm not sure if it's safe to modify the existing constructor; can we overload it instead?

Reply:
I just think it's private, but it is more safe that use overload instead of modify it. Thank you.